fix: webauthn4j を 0.31.5 にバンプして FIDO MDS BLOB v242 パースエラーを解消

## 概要

FIDO2 登録時に毎回 `Failed to fetch MDS BLOB from FIDO Alliance: Input data does not match expected form` の ERROR ログが出ている。原因は webauthn4j のバージョンが古く、FIDO Alliance が配信する最新の MDS BLOB (v242) を解釈できないこと。

## 症状

`enrichWithMdsMetadata` 呼び出しのたびに以下のスタックトレースが ERROR レベルで出力される（リクエスト毎）。

```
ERROR Failed to fetch MDS BLOB from FIDO Alliance: Input data does not match expected form
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: value is out of range
  (through reference chain: ArrayList[83]->ArrayList[4])
  at com.webauthn4j.converter.jackson.deserializer.json.AttachmentHintFromStringDeserializer.deserialize
```

## 影響範囲

- **登録自体は成功**（`enrichWithMdsMetadata` は try-catch で fail-open）
- 認証器メタデータ（`authenticator_name`, `is_compromised` 等）が credential metadata に付与されない
- 認証器のセキュリティチェック（compromised 検出）が機能しない
- ERROR ログがリクエスト毎に出力され、監視・アラートのノイズになる
- 失敗したフェッチがキャッシュに保存されないため、毎リクエストで再フェッチが走る（~14秒の遅延）

## 根本原因

- 現状: `com.webauthn4j:webauthn4j-core:0.30.0.RELEASE`
- FIDO Alliance の MDS BLOB v242 で `AttachmentHint` 列挙値が追加された
- 0.30.0 にはその値の対応がなく、Jackson デシリアライズ時に `InvalidFormatException`
- webauthn4j 0.30.3 / 0.31.3 以降で **FIDO MDS Blob v242 サポート** が追加されている

## 修正方針

`webauthn4j-core` / `webauthn4j-metadata` を **0.31.5.RELEASE** にバンプ。

### 対象ファイル

- \`libs/idp-server-webauthn4j-adapter/build.gradle\`
  - \`webauthn4j-core: 0.30.0 → 0.31.5\`
  - \`webauthn4j-metadata: 0.30.0 → 0.31.5\`
- \`libs/idp-server-springboot-adapter/build.gradle\`
  - \`webauthn4j-core: 0.28.5 → 0.31.5\`（バージョンも統一）

### 動作確認結果（ローカル）

0.31.5 で Docker Compose 再起動後、FIDO2 登録試行：

```
INFO  Fetching MDS BLOB from FIDO Alliance...
INFO  MDS fetched and cached with 323 entries, nextUpdate: 2026-06-01
INFO  webauthn4j registration succeeded
```

ERROR は出ず、323 エントリのキャッシュ生成成功、credential に \`is_compromised: false\` も付与された。

## 関連する改善余地（別 Issue 化候補）

- [ ] MDS フェッチ失敗時のログレベルを ERROR → WARN に（fail-open 設計のため）
- [ ] 失敗時のネガティブキャッシュ（短い TTL）を入れて再フェッチ嵐を防ぐ
- [ ] fail-open / fail-closed のポリシー要件整理（商用要件次第）
- [ ] 2 モジュール間で webauthn4j バージョンが不一致だった点の再発防止（dependency-version の集中管理）

## 備考

webauthn4j 0.31.x シリーズは Jackson 3 + JDK 17 ベースラインで、現状の Spring Boot 4 + Jackson 3 + JDK 21 構成と互換性あり。0.30.0 → 0.31.5 へのバンプで idp-server 側の API 利用箇所はコンパイル通過済（deprecated API warning が \`MdsCacheEntry.java\` に 1 件あるが致命的でない）。

--- a/libs/idp-server-springboot-adapter/build.gradle
+++ b/libs/idp-server-springboot-adapter/build.gradle
@@ -39,7 +39,7 @@ dependencies {
 	implementation 'org.springframework.session:spring-session-core'
 
 	//webauthn
-	implementation 'com.webauthn4j:webauthn4j-core:0.28.5.RELEASE'
+	implementation 'com.webauthn4j:webauthn4j-core:0.31.5.RELEASE'
 
 	developmentOnly 'org.springframework.boot:spring-boot-devtools'
 

--- a/libs/idp-server-webauthn4j-adapter/build.gradle
+++ b/libs/idp-server-webauthn4j-adapter/build.gradle
@@ -15,8 +15,8 @@ dependencies {
 	implementation project(':libs:idp-server-platform')
 	implementation project(':libs:idp-server-core')
 	implementation project(':libs:idp-server-authentication-interactors')
-	implementation 'com.webauthn4j:webauthn4j-core:0.30.0.RELEASE'
-	implementation 'com.webauthn4j:webauthn4j-metadata:0.30.0.RELEASE'
+	implementation 'com.webauthn4j:webauthn4j-core:0.31.5.RELEASE'
+	implementation 'com.webauthn4j:webauthn4j-metadata:0.31.5.RELEASE'
 
 	testImplementation platform('org.junit:junit-bom:5.10.0')
 	testImplementation 'org.junit.jupiter:junit-jupiter'

